barretenberg/cpp/src/barretenberg/crypto/pedersen_hash/pedersen_refactor.cpp

```diff
@@ -0,0 +1,57 @@
+#include "./pedersen_refactor.hpp"
+#include <iostream>
+#ifndef NO_OMP_MULTITHREADING
+#include <omp.h>
+#endif
+
+// TODO(@zac-wiliamson #2341 rename to pedersen.cpp once we migrate to new hash standard)
+
+namespace crypto {
+
+using namespace generators;
+
+/**
+ * Given a vector of fields, generate a pedersen hash using the indexed generators.
+ */
+
+/**
+ * @brief Given a vector of fields, generate a pedersen hash using generators from `generator_context`.
+ *
+ * @details `hash_index` is used to access offset elements of `generator_context` if required.
+ *          e.g. if one desires to compute
+ *          `inputs[0] * [generators[hash_index]] + `inputs[1] * [generators[hash_index + 1]]` + ... etc
+ *          Potentially useful to ensure multiple hashes with the same domain separator cannot collide.
+ *
+ * TODO(@suyash67) can we change downstream code so that `hash_index` is no longer required? Now we have a proper
+ * domain_separator parameter, we no longer need to specify different generator indices to ensure hashes cannot collide.
```

Comment:
@suyash67 -- my thoughts are that if this is for hashing and not for committing, then this should be okay

Reply:
I'll fix this in the next PR. Requires changing all the c_binds, will be messy